huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go

```diff
@@ -806,19 +806,38 @@ func resourceGeminiDBInstanceV3Update(d *schema.ResourceData, meta interface{})
 			}
 		}
 		if newnum.(int) < old.(int) {
-			//Reduce Nodes
+			// Reduce Nodes
 			shrinkSize := old.(int) - newnum.(int)
-			reduceNodeOpts := instances.ReduceNodeOpts{
-				Num: 1,
+			// API accepts maxinum num of 10
+			reduceNum := 10
+			loopSize := shrinkSize / 10
+			lastNum := shrinkSize % 10
+			if lastNum > 0 {
+				loopSize += 1
 			}
-			logp.Printf("[DEBUG] Reduce Node Options: %+v", reduceNodeOpts)
 
-			for i := 0; i < shrinkSize; i++ {
-				result := instances.ReduceNode(client, d.Id(), reduceNodeOpts)
-				if result.Err != nil {
-					return fmtp.Errorf("Error shrinking huaweicloud_gaussdb_cassandra_instance %s node size: %s", d.Id(), result.Err)
+			for i := 0; i < loopSize; i++ {
+				if lastNum > 0 && (i == loopSize-1) {
+					reduceNum = lastNum
+				}
+				reduceNodeOpts := instances.ReduceNodeOpts{
+					Num: reduceNum,
+				}
+				logp.Printf("[DEBUG] Reduce Node Options: %+v", reduceNodeOpts)
+
+				n, err := instances.ReduceNode(client, d.Id(), reduceNodeOpts).Extract()
+				if err != nil {
+					return fmtp.Errorf("Error shrinking huaweicloud_gaussdb_cassandra_instance %s node size: %s", d.Id(), err)
```

Comment:
we should remove `huaweicloud` word from our error msg and start with a lowercase letter.

Reply:
done